darwin/dataset/remote_dataset.py

```diff
@@ -251,16 +255,46 @@ def pull(
             If the given ``release`` has an invalid format.
         ValueError
             If darwin in unable to get ``Team`` configuration.
+        ValueError
+            If the release is still processing after the maximum retry duration.
         """
 
         console = self.console or Console()
 
         if release is None:
-            release = self.get_release()
+            if retry:
+                raise ValueError(
+                    "To retry downloading a release, a release name must be provided. This can be done as follows:\n\nrelease = dataset.get_release(name='release_name')\ndataset.pull(release=release, retry=True)"
+                )
+            else:
+                release = self.get_release(retry=retry)
 
         if release.format != "json" and release.format != "darwin_json_2":
             raise UnsupportedExportFormat(release.format)
 
+        if release.status == "pending":
+            if retry:
+                retry_duration = 300
+                retry_interval = 10
+                while release.status == "pending" and retry_duration > 0:
```

Comment:
Poll every 10 seconds, up to a maximum of 5 minutes. I'm open to feedback on what the duration of `retry_duration` should be!